core/mpc/dkls/dkls.ts

```diff
@@ -226,13 +227,13 @@ export class DKLS {
     for (let i = 0; i < 3; i++) {
       try {
         const result = await this.startKeygen(i)
-        if (result !== undefined) {
-          return result
-        }
+        return result
       } catch (error) {
         console.error('DKLS keygen error:', error)
+        throw new Error('DKLS keygen failed')
```

Comment:
this is not correct , because the logic is to retry the keygen up to 3 times when there is an error , 
This change will get rid of the retry logic